File: 70-climbing-stairs/climbing-stairs.js
/**
 * @param {number} n
 * @return {number}
 */
var climbStairs = function(n) {
    if (n === 1) return 1;
    if (n === 2) return 2;

    let prev1 = 2, prev2 = 1;

    for (let i = 3; i <= n; i++) {
        let v = prev2 + prev1
        prev2 = prev1
        prev1 = v
    }
    return prev1
};

File: 70-climbing-stairs/climbing-stairs.java
class Solution {
    public int climbStairs(int n) {
        int[] dp = new int[n+1];
        dp[n] = 1;

        for (int i = n-1; i >= 0; i--) {
            int ways = 0;
            if (i+1 <= n) {
                ways += dp[i+1];
            }
            if (i+2 <= n) {
                ways += dp[i+2];
            }
            dp[i] = ways;
        }

        return dp[0];
    }
}